core/node/api_server/src/web3/state.rs

```diff
@@ -241,6 +247,51 @@ impl SealedL2BlockNumber {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct BridgeAddressesHandle(Arc<RwLock<BridgeAddresses>>);
+
+impl BridgeAddressesHandle {
+    pub fn new(bridge_addresses: BridgeAddresses) -> Self {
+        Self(Arc::new(RwLock::new(bridge_addresses)))
+    }
+
+    /// Creates a handle to the bridge addresses together with a task that will update them on a schedule.
+    pub fn new_with_updater(
+        main_node_client: Box<DynClient<L2>>,
+        initial_value: BridgeAddresses,
+        update_interval: Duration,
+        stop_receiver: watch::Receiver<bool>,
+    ) -> (Self, impl Future<Output = anyhow::Result<()>>) {
+        let this = Self(Arc::new(RwLock::new(initial_value)));
+        let updater = this.clone();
+
+        let update_task = async move {
+            loop {
+                if *stop_receiver.borrow() {
```

Comment:
Nit: You can use `while !*stop_receiver.borrow()` as the loop statement instead, and move the log after the loop.